cdap-ui/app/cdap/components/DataPrep/TopPanel/AddToPipelineModal.js

```diff
@@ -335,6 +335,47 @@ export default class AddToHydratorModal extends Component {
     };
   }
 
+  constructBigQuerySource(artifactsList, bigqueryInfo) {
```

Comment:
Is the only difference between this function and the one in `PipelineConfigHelper` the error handling for `if (!batchArtifact)`? Can/Should we try to combine them?